core/tests/behavior/async_write.rs

```diff
@@ -674,3 +675,31 @@ pub async fn test_write_with_if_not_exists(op: Operator) -> Result<()> {
 
     Ok(())
 }
+
+/// Write an file with if_match will get a ConditionNotMatch error if file's etag does not match.
+pub async fn test_write_with_if_match(op: Operator) -> Result<()> {
+    if !op.info().full_capability().write_with_if_match {
+        return Ok(());
+    }
+
+    let (path, content, _) = TEST_FIXTURE.new_file(op.clone());
+    let (path_incorrect, _, _) = TEST_FIXTURE.new_file(op.clone());
+
+    op.write(&path, content.clone()).await?;
+    op.write(&path_incorrect, content.clone()).await?;
```

Comment:
I'm guessing `path_incorrect` should write data `content_incorrect`? Otherwise they are write the same data. We cah test in this way:


```
let (path_a, content_a, _) = TEST_FIXTURE.new_file(op.clone());
let (path_b, content_b, _) = TEST_FIXTURE.new_file(op.clone());

write_with(path_a, content_a).if_match(etag_a) -> should succeed.
write_with(path_a, content_a).if_match(etag_b) -> should fail.
```